src/main/java/com/pmm/budgetcalculator/expensing/service/ExpenseService.java

```diff
@@ -20,15 +22,30 @@ public List<Expense> getAllExpenses() {
     public Expense getExpenseById(Long id) {
         return expenseRepository
                 .findById(id)
-                .orElseThrow(() -> new RuntimeException("Expense not found for Id:" + id));
+                .orElseThrow(() -> new ExpenseNotFoundException(id));
     }
 
     public Expense createExpense(Expense expense) {
         return expenseRepository.save(expense);
     }
-        public void removeExpenseById(Long id){
-            if (expenseRepository.existsById(id)) {
-                expenseRepository.deleteById(id);
-            }
+
+    public void removeExpenseById(Long id) {
+        if (expenseRepository.existsById(id)) {
+            expenseRepository.deleteById(id);
+        } else ResponseEntity.notFound().build();
+    }
+
+    public ResponseEntity<Expense> updateExpense(Long id, Expense expenseDetails) {
```

Comment:
Do it like every other method. Return Expense and in controller wrap in ResponseEntity<>